File: src/main/java/io/github/symmetricdevs/supercritical/core/mixin/gregtech/MixinIArmorLogic.java
package io.github.symmetricdevs.supercritical.core.mixin.gregtech;

import com.gregtechceu.gtceu.api.item.armor.IArmorLogic;
import org.spongepowered.asm.mixin.Mixin;
import io.github.symmetricdevs.supercritical.api.items.armor.ArmorLogicExtension;

/**
 * Injects {@link ArmorLogicExtension} into GTCEu's armor logic interface.
 */
@Mixin(value = IArmorLogic.class, remap = false)
public interface MixinIArmorLogic extends ArmorLogicExtension {
}

File: src/main/kotlin/supercritical/core/mixin/gregtech/MixinIArmorLogic.kt
package supercritical.core.mixin.gregtech

import com.gregtechceu.gtceu.api.item.armor.IArmorLogic
import org.spongepowered.asm.mixin.Mixin
import supercritical.api.items.armor.ArmorLogicExtension

/**
 * Injects [ArmorLogicExtension] into GTCEu's armor logic interface.
 */
@Mixin(value = [IArmorLogic::class], remap = false)
interface MixinIArmorLogic : ArmorLogicExtension
